gradle-incremental-configuration-cache/src/test/groovy/com/palantir/gradle/configcache/incremental/IncrementalConfigurationCacheTest.groovy

```diff
@@ -35,104 +34,102 @@ class IncrementalConfigurationCacheTest extends IntegrationTestKitSpec {
                 mavenCentral()
                 mavenLocal()
             }
-        """.stripIndent(true)
+        '''.stripIndent(true)
 
-        expect:
-        assert runAndAssertNoConfigurationCache("classes")
+        when:
+        def buildResult = createRunner('classes', '--configuration-cache').buildAndFail()
+
+        then:
+        buildResult.output.contains('Configuration cache allowed tasks file not found')
     }
 
-    def "tasks in allowlist run with config cache"() {
-        // Save the task graph of `build` to the allow list file
-        // language=gradle
-        buildFile << """
-        import com.palantir.gradle.configcache.incremental.IncrementalConfigurationCachePlugin
-        import java.nio.file.Files
-
-        apply plugin: 'java-library'
-        
-        def allowListFilePath = getProjectDir().toPath().resolve(IncrementalConfigurationCachePlugin.ALLOW_LIST_FILE)
-        Files.createDirectories(allowListFilePath.getParent())
-        def allowListFile = file(allowListFilePath)
-        allowListFile.createNewFile()
-        println("File created at: " + allowListFilePath)
-
-        gradle.taskGraph.whenReady { graph ->
-            graph.getAllTasks().each { task ->
-                 allowListFile << task.path
-                 allowListFile << '\\n'
-            }
-        }
-        """
-        assert createRunner("classes").build().output.contains("BUILD SUCCESSFUL")
-        buildFile.text = ''
+    def "blows up if applied to non root project"() {
+        def subprojectDir = directory('subproject')
+        file('build.gradle', subprojectDir) << '''
+            apply plugin: 'com.palantir.incremental-configuration-cache'
+            apply plugin: 'java-library'
+        '''.stripIndent(true)
+
+        directory("gradle")
+        file("gradle/configuration-cache-allowed-tasks") << '''
+            :compileJava
+            :processResources
+            :classes
+        '''.stripIndent(true)
 
-        // Run the main build
         // language=gradle
-        buildFile << """
-            import com.palantir.gradle.configcache.incremental.IncrementalConfigurationCachePlugin
+        settingsFile << '''
+            include 'subproject'
+        '''.stripIndent(true)
 
+        // language=gradle
+        buildFile << '''
             apply plugin: 'com.palantir.incremental-configuration-cache'
             apply plugin: 'java-library'
 
             repositories {
                 mavenCentral()
                 mavenLocal()
```

Comment:
I don't think we need mavenLocal() here? Do we need this repositories block at all? I don't think we're downloading any dependencies in the tests, just using stuff built into Gradle.